src/test/java/com/segment/analytics/android/integration/intercom/IntercomTest.java

```diff
@@ -1 +1,181 @@
+
+        verifyStatic();
+        Intercom.initialize(application, "123", "123");
+    }
+
+    @Test
+    public void identify() {
+        Traits traits = createTraits("123");
+        integration.identify(new IdentifyPayloadBuilder().traits(traits).build());
+
+        ArgumentCaptor<Registration> argument = ArgumentCaptor.forClass(Registration.class);
+        verify(intercom).registerIdentifiedUser(argument.capture());
+        assertEquals("123", argument.getValue().getUserId());
+    }
+
+    @Test
+    public void identifyWithUserHash() {
+        Traits traits = createTraits("123");
+        Options options = new Options();
+        Map<String, Object> intercomOptions = new HashMap<>();
+        intercomOptions.put("userHash", "567");
+        options.setIntegrationOptions("Intercom", intercomOptions);
+        integration.identify(new IdentifyPayloadBuilder().traits(traits).options(options).build());
+        verify(intercom).setUserHash("567");
+    }
+
+    @Test
+    public void identifyWithSpeccedAttributes() {
+        long createdAt = 123344L;
+        Traits traits = createTraits("123");
+        traits.putName("Brennan");
+        traits.putEmail("testing@segment.com");
+        traits.putPhone("1112223333");
+        Options options = new Options();
+        Map<String, Object> intercomOptions = new HashMap<>();
+        intercomOptions.put("languageOverride", "testing");
+        intercomOptions.put("createdAt", createdAt);
+        intercomOptions.put("unsubscribedFromEmails", true);
+        options.setIntegrationOptions("Intercom", intercomOptions);
+        integration.identify(new IdentifyPayloadBuilder().traits(traits).options(options).build());
+
+        ArgumentCaptor<UserAttributes> attributesArgumentCaptor = ArgumentCaptor.forClass(UserAttributes.class);
+        verify(intercom).updateUser(attributesArgumentCaptor.capture());
+
+        UserAttributes capturedAttributes = attributesArgumentCaptor.getValue();
+        Map<String, Object> mapOfAttributes = capturedAttributes.toMap();
+
+        assertEquals("Brennan", String.valueOf(mapOfAttributes.get("name")));
+        assertEquals("testing@segment.com", String.valueOf(mapOfAttributes.get("email")));
+        assertEquals("1112223333", String.valueOf(mapOfAttributes.get("phone")));
+        assertEquals("testing", String.valueOf(mapOfAttributes.get("language_override")));
+        assertEquals(createdAt, mapOfAttributes.get("signed_up_at"));
+        assertEquals(true, mapOfAttributes.get("unsubscribed_from_emails"));
+    }
+
+    @Test
+    public void identifyWithCompany() {
+        Traits traits = createTraits("123");
+        Map<String, Object> company = new HashMap<>();
+        company.put("id", "456");
+        company.put("name", "Acme");
+        traits.put("company", company);
+        integration.identify(new IdentifyPayloadBuilder().traits(traits).build());
+        verify(intercom).updateUser(any(UserAttributes.class));
```

Comment:
Agreed - awaiting Intercom to let me know whether they can make the "getAttributes" method in Company public - otherwise there's no attributes getting for this Class.